src/Workspaces/CSharp/Portable/Formatting/CSharpFormattingOptions.cs

```diff
@@ -1,148 +1,351 @@
-            storageLocations: new RoamingProfileStorageLocation("TextEditor.CSharp.Specific.SpaceWithinCastParentheses"));
+            storageLocations: new OptionStorageLocation[] {
+                new EditorConfigStorageLocation("csharp_space_within_parentheses", (s) => ParseSpaceOptions(s, SpacingOption.TypeCasts)),
+                new RoamingProfileStorageLocation("TextEditor.CSharp.Specific.SpaceWithinCastParentheses")});
 
         public static Option<bool> SpaceWithinOtherParentheses { get; } = new Option<bool>(nameof(CSharpFormattingOptions), nameof(SpaceWithinOtherParentheses), defaultValue: false,
-            storageLocations: new RoamingProfileStorageLocation("TextEditor.CSharp.Specific.SpaceWithinOtherParentheses"));
+            storageLocations: new OptionStorageLocation[] {
+                new EditorConfigStorageLocation("csharp_space_within_parentheses", (s) => ParseSpaceOptions(s, SpacingOption.ControlFlowStatements)),
+                new RoamingProfileStorageLocation("TextEditor.CSharp.Specific.SpaceWithinOtherParentheses")});
+
+        private static object ParseSpaceOptions(string value, SpacingOption parenthesesSpacingOption)
+            => (from v in value.Split(',').Select(v => v.Trim())
+                let option = ConvertToSpacingOption(v)
+                where option != null
+                select option.Value)
+                .Any(v => v == parenthesesSpacingOption);
+
+        private static SpacingOption? ConvertToSpacingOption(string v)
+        {
+            switch (v)
+            {
+                case "expressions": return SpacingOption.Expressions;
+                case "type_casts": return SpacingOption.TypeCasts;
+                case "control_flow_statements": return SpacingOption.ControlFlowStatements;
+                default: return null;
+            }
+        }
 
         public static Option<bool> SpaceAfterCast { get; } = new Option<bool>(nameof(CSharpFormattingOptions), nameof(SpaceAfterCast), defaultValue: false,
-            storageLocations: new RoamingProfileStorageLocation("TextEditor.CSharp.Specific.SpaceAfterCast"));
+            storageLocations: new OptionStorageLocation[] {
+                new EditorConfigStorageLocation("csharp_space_after_cast"),
+                new RoamingProfileStorageLocation("TextEditor.CSharp.Specific.SpaceAfterCast")});
 
         public static Option<bool> SpacesIgnoreAroundVariableDeclaration { get; } = new Option<bool>(nameof(CSharpFormattingOptions), nameof(SpacesIgnoreAroundVariableDeclaration), defaultValue: false,
-            storageLocations: new RoamingProfileStorageLocation("TextEditor.CSharp.Specific.SpacesIgnoreAroundVariableDeclaration"));
+            storageLocations: new OptionStorageLocation[] {
+                new EditorConfigStorageLocation("csharp_space_in_declaration_statements"),
```

Comment:
One of these says "spacesignore" and the other one says "space_in" -- is this backwards?